### PR TITLE
Improve: Dockerfile and Pipeline

### PR DESCRIPTION
**Pull request by EXONER4TED**
Signed-off-by: Logan Bond <lbond@secureworks.com>

**What type of PR is this?**

> Uncomment one (or more) `/kind <>` lines:

> /kind bug

/kind cleanup

> /kind documentation

> /kind feature

**Any specific area of the project related to this PR?**

> Uncomment one (or more) `/area <>` lines:

> /area crawler

/area ci

> /area utils

**What this PR does / why we need it**:
Updates the Dockerfile to be in the root of the repo, and it removes the shell script wrapper from the Dockerfile. The shell script in the Dockerfile can obfuscate errors as the failure reported out in the container logs may just report the line in the shell script, not the actual python.

I also updated/improved the Actions side of the CI pipeline - mainly just because I like pipelines and these were low hanging fruit 🙂 . The update adds a couple checks:
1. Linting of the Dockerfile, YAML files, and python code (python is set to allow failure for now)
2. A test build of the Dockerfile that doesn't get pushed anywhere - just to check that it builds

These checks should be required to pass before anyone can get a branch merged in. It's mostly just for ensuring the new code coming in builds in the container correctly and looks nice 🙂 

Once lint has been fixed in the python code (later), a check could be easily added that the python package builds as well. But I didn't want to add linting the repo to the scope of this PR.

OH, and I updated the container to run as a non-root user.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
               --platform "arm64,amd64" \
               -t falcosecurity/kernel-crawler:${CIRCLE_TAG} \
               -t falcosecurity/kernel-crawler:latest \
-              -f docker/Dockerfile .
+              -f Dockerfile .
```
Comment (FedeDP) [inline .circleci/config.yml]: I'd leave the docker folder indeed :)

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -1,35 +1,78 @@
...
+
+      - name: Lint YAML
+        uses: karancode/yamllint-github-action@v2.0.0
+        with:
+          yamllint_file_or_dir: '.github/workflows/ci.yml'
+          yamllint_strict: false
+          yamllint_comment: false
+
+  build:
+    name: Build
```
Comment (FedeDP) [inline .github/workflows/ci.yml]: It's ok but i'd use a path-filter to only build it if docker/ subfolder was changed. See https://github.com/falcosecurity/libs/blob/master/.github/workflows/ci.yml#L11 for an example :)

I'd also change the job name to "build Dockerfile"

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,16 @@
+FROM python:3.8-slim-buster
```
Comment (FedeDP) [inline Dockerfile]: On master i switched to `python:3.10-slim` :)

File: .circleci/config.yml (hunk 1 above)
Comment (EXONER4TED) [reply .circleci/config.yml]: This is not how most projects do it, but that's fine. I've adjusted the Action to use the folder but keep the build context top level.

File: .github/workflows/ci.yml (hunk 2 above)
Comment (EXONER4TED) [reply .github/workflows/ci.yml]: path-filter is a hack 🙃 I just broke out the files to use the native filtering.

File: Dockerfile (hunk 3 above)
Comment (EXONER4TED) [reply Dockerfile]: 👍

**Comment by FedeDP** [comment]
I will need to find the time to test that the new docker image is exactly equivalent with the old ones :) 
But this LGTM! Thank you Logan!

**Comment by FedeDP** [comment]
Hi! Can you rebase this PR?

**Comment by EXONER4TED** [comment]
>Hi! Can you rebase this PR?

Better late than never? 😅

**Comment by FedeDP** [comment]
> Better late than never? sweat_smile

Yep! Thank you for this great contribution! :)

**Review by FedeDP — APPROVED** [review]
/approve

**Pull request merged**